# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/HTMLEditSubActionHandler.cpp
# Commit: 20512741e7b6
# Full Hash: 20512741e7b60f3f6791002d8311d681f9242e68
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2023-02-02 09:15:42
# Description:
#   Bug 1812170 - Make `HTMLEditor::ClearStyleAt` should return `pointToPutCaret` if next node of first split is not split r=m_kato
#   
#   The new path added in bug 1807829 hits the odd result case of `ClearStyleAt`.
#   `pointToPutCaret` may be updated if the preceding split occurs.  Therefore,
#   it should be returned if it does not split next nodes after the first split
# ==============================================================================

diff -r d9138ddf3c3d -r 20512741e7b6 editor/libeditor/HTMLEditSubActionHandler.cpp
--- a/editor/libeditor/HTMLEditSubActionHandler.cpp	Thu Feb 02 02:25:04 2023 +0000
+++ b/editor/libeditor/HTMLEditSubActionHandler.cpp	Thu Feb 02 05:09:15 2023 +0000
@@ -6108,6 +6108,7 @@
         return pointToPutCaretOrError;
       }
       pointToPutCaret = pointToPutCaretOrError.unwrap();
+      MOZ_ASSERT(pointToPutCaret.IsSet());
       pendingStyle = mPendingStylesToApplyToNewContent->TakeClearingStyle();
     }
   }